### PR TITLE
Use database for searches and storage

### PR DESCRIPTION
**Pull request by brandonxia01**
This PR adds:
* Uses the SQLite/WebSQL databases to query songs from the DB to store and use in lyric mode.
* Lyric based search. Retrieves matches from the DB and then sorts them by how close the title and lyrics are to the search string.

Searching for lyrics:
Screen Shot 2020-12-26 at 6 25 58 PM

----------------------------------------------------

Search hits sorted by title and lyric accuracy. But since title is much shorter, matches in the title weigh more, so they usually come first.
Screen Shot 2021-01-01 at 3 01 30 AM

### CODE REVIEW

**Review by brotatotes — COMMENTED** [review]
Changes look really good! 

I'm incredibly wary about the lagging typing change. I think we should figure that out before we check in this new behavior.

**Comment by brotatotes** [comment]
I think you may have mentioned this already, but probably to update/add some unit tests as appropriate for this change as well.

**Comment by brandonxia01** [comment]
> I'm incredibly wary about the lagging typing change. I think we should figure that out before we check in this new behavior.

Yeah I agree, I'll take another look at it tomorrow. Maybe i'll post a question on stackoverflow lol

File: src/pages/BookPage.tsx [hunk 1]
```diff
@@ -31,17 +48,20 @@ const BookPage: React.FC = () => {
         <NavigationBar backButtonOnClick={() => history.push("/")} />
       </IonHeader>
       <IonItem>
-        <IonSearchbar
+        <IonInput
```
Comment (brotatotes) [inline src/pages/BookPage.tsx:51]: OH, when you said you changed it to input I thought you meant HTML input element. Very happy to see we are still able to get this working with an Ion element here. :)

File: src/test/utils/SecurityUtils.test.tsx [hunk 2]
```diff
@@ -2,7 +2,7 @@ import { decrypt, encrypt } from "../../utils/SecurityUtils";
 
 describe("Security Utils Tests", () => {
   it("Encrypt and Decrypt Test", () => {
-    const text = JSON.stringify({ data: "hello world" });
```
Comment (brotatotes) [inline src/test/utils/SecurityUtils.test.tsx:5]: Does this (and other changes to SecurityUtils) fix the flakiness?

**Review by brotatotes — APPROVED** [review]
N I C E

File: src/pages/BookPage.tsx (hunk 1 above)
Comment (brandonxia01) [reply src/pages/BookPage.tsx:51]: Ah yeah at first I did make it `<input>` but then I realized `IonSearchbar != IonInput` so we can stay keep ion here.

File: src/test/utils/SecurityUtils.test.tsx (hunk 2 above)
Comment (brandonxia01) [reply src/test/utils/SecurityUtils.test.tsx:5]: Yeah 🤞 not sure why I had it like this in the first place.

**Pull request merged**